onboarding/mlsys/1-backprop.md

```diff
@@ -0,0 +1,414 @@
+# Part 1: Backpropagation
+
+Machine learning is all about getting computers to solve math problems for you.
+
+Specifically, it's about solving math problems like, "For a given mathematical
+model, what weights in the model would cause the model to calculate the most
+accurate predictions for a set of inputs?"
+
+You're going to write code to solve this exact problem for one particularly
+simple model.
+
+Afterwards, you will understand of basic concepts like gradient descent and
+batch size.  You will also understand how those basic concepts translate into
+some of Determined's advanced features, like gradient aggregation,
+data-parallel distributed training, and model-parallel distributed training.
+
```

Comment:
I want to add a section on definitions here at the top before they start the exercise. This will help orient the reader to the concepts.

## Terms and Definitions

Gradient Descent.
Definition.

Batch Size.
Definition.

Gradient Aggregation.
Definition.

Data-Parallel Distributed Training.
Definition.

Model-Parallel Distributed Training.
Definition.

Reply:
We should sit down and you can show me what this would look like, because from my perspective, the whole doc is mostly definitions of terms.